src/doc_builder/build_embeddings.py

```diff
@@ -0,0 +1,439 @@
+            embeddings = Embedding(
+                text=chunk.text, source=chunk.source, package_name=chunk.package_name, embedding=_embedding
+            )
+            embeddings.append(embeddings)
+
+    return embeddings
+
+
+def build_embeddings(
+    package_name,
+    doc_folder,
+    version="main",
+    version_tag="main",
+    language="en",
+    is_python_module=False,
+    version_tag_suffix="src/",
+    repo_owner="huggingface",
+    repo_name=None,
+):
+    """
+    Build the documentation of a package.
+
+    Args:
+        package_name (`str`): The name of the package.
+        doc_folder (`str` or `os.PathLike`): The folder in which the source documentation of the package is.
+        version (`str`, *optional*, defaults to `"main"`): The name of the version of the doc.
+        version_tag (`str`, *optional*, defaults to `"main"`): The name of the version tag (on GitHub) of the doc.
+        language (`str`, *optional*, defaults to `"en"`): The language of the doc.
+        is_python_module (`bool`, *optional*, defaults to `False`):
+            Whether the docs being built are for python module. (For example, HF Course is not a python module).
+        version_tag_suffix (`str`, *optional*, defaults to `"src/"`):
+            Suffix to add after the version tag (e.g. 1.3.0 or main) in the documentation links.
+            For example, the default `"src/"` suffix will result in a base link as `https://github.com/huggingface/{package_name}/blob/{version_tag}/src/`.
+            For example, `version_tag_suffix=""` will result in a base link as `https://github.com/huggingface/{package_name}/blob/{version_tag}/`.
+        repo_owner (`str`, *optional*, defaults to `"huggingface"`):
+            The owner of the repository on GitHub. In most cases, this is `"huggingface"`. However, for the `timm` library, the owner is `"rwightman"`.
+        repo_name (`str`, *optional*, defaults to `package_name`):
+            The name of the repository on GitHub. In most cases, this is the same as `package_name`. However, for the `timm` library, the name is `"pytorch-image-models"` instead of `"timm"`.
+    """
+    page_info = {
+        "version": version,
+        "version_tag": version_tag,
+        "language": language,
+        "package_name": package_name,
+        "repo_owner": repo_owner,
+        "repo_name": repo_name if repo_name is not None else package_name,
+    }
+
+    read_doc_config(doc_folder)
+
+    package = importlib.import_module(package_name) if is_python_module else None
+
+    # Step 1: create chunks
+    chunks = create_chunks(
+        package,
+        doc_folder,
+        page_info,
+        version_tag_suffix=version_tag_suffix,
+        is_python_module=is_python_module,
+    )
+
+    # Step 2: create embeddings
+    embeddings = call_embed_api(chunks)
+    print(len(embeddings))
+
+    # Step 3: push embeddings to vector database
+    # TODO
```

Comment:
ping again when you have this part i'd be interested in seeing it

Reply:
https://github.com/huggingface/doc-builder/pull/506#issuecomment-2176411230